socs/agents/srs_cg635m/agent.py

```diff
@@ -0,0 +1,198 @@
+import argparse
+import socket
+import time
+from os import environ
+
+import txaio
+from ocs import ocs_agent, site_config
+from ocs.ocs_twisted import TimeoutLock
+
+from socs.agents.srs_cg635m.drivers import SRS_CG635m_Interface
+
+
+class SRSCG635mAgent:
+    def __init__(self, agent, ip_address, gpib_slot):
+        self.agent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+
+        self.job = None
+        self.ip_address = ip_address
+        self.gpib_slot = gpib_slot
+        self.monitor = False
+
+        self.clock = None
+
+        # Registers Temperature and Voltage feeds
+        agg_params = {
+            'frame_length': 10 * 60,
+        }
+        self.agent.register_feed('srs_clock',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=0)
+
+    @ocs_agent.param('auto_acquire', default=False, type=bool)
+    def init(self, session, params=None):
+        """init(auto_acquire=False)
+
+        **Task** - Initialize the connection to the srs clock.
+
+        Parameters
+        ----------
+        auto_acquire: bool, optional
+            Default is False. Starts data acquisition after initialization
+            if True.
+        """
+        with self.lock.acquire_timeout(0) as acquired:
+            if not acquired:
+                return False, "Could not acquire lock"
+
+            try:
+                self.clock = SRS_CG635m_Interface(self.ip_address, self.gpib_slot)
+                self.idn = self.clock.identify()
+
+            except socket.timeout as e:
+                self.log.error(f"Clock timed out during connect: {e}")
+                return False, "Timeout"
+            self.log.info("Connected to Clock: {}".format(self.idn))
+
+        # Start data acquisition if requested in site-config
+        auto_acquire = params.get('auto_acquire', False)
+        if auto_acquire:
+            self.agent.start('acq')
+
+        return True, 'Initialized Clock.'
+
+    @ocs_agent.param('test_mode', default=False, type=bool)
+    @ocs_agent.param('wait', default=1, type=float)
+    def acq(self, session, params):
+        """acq(wait=1, test_mode=False)
+
+        **Process** - Continuously monitor srs clock lock registers
+        and send info to aggregator.
```

Comment:
Replace "send info to aggregator" with "publish to a feed." The current sentence doesn't really capture that it's also going to other subscribers, like the InfluxDB publisher agent, i.e. to Grafana.